src/logd.rs

```diff
@@ -0,0 +1,117 @@
+use std::{io, os::unix::net::UnixDatagram, path::Path, time::SystemTime};
+
+use bytes::BufMut;
+use parking_lot::RwLockUpgradableReadGuard;
+
+use crate::{thread, Buffer, Event, Priority, LOGGER_ENTRY_MAX_LEN};
+
+/// Logd write socket path
+const LOGDW: &str = "/dev/socket/logdw";
+
+lazy_static::lazy_static! {
+    static ref SOCKET: LogdSocket = LogdSocket::connect(Path::new(LOGDW));
+}
+
+/// Logd write socket abstraction. Sends never fail and on each send a reconnect
+/// reconnect attempt is made.
+struct LogdSocket {
+    socket: parking_lot::RwLock<UnixDatagram>,
+}
+
+impl LogdSocket {
+    /// Construct a new LogdSocket.
+    ///
+    /// # Panics
+    ///
+    /// This function panics when the socket cannot be created.
+    pub fn connect(path: &Path) -> LogdSocket {
```

Comment:
shouldn't these methods be private?

Reply:
`connect` is called from the `lazy_static` init and therefore must be `pub`.